src/screens/reader/components/ChapterDrawer/index.tsx

```diff
@@ -65,129 +68,70 @@ const ChapterDrawer = ({
     },
   });
 
-  const changeChapter = (item: ChapterInfo) => {
-    navigation.replace('Chapter', {
-      novel: novelItem,
-      chapter: item,
-    });
-  };
-  const renderItem: FlashListProps<ChapterInfo>['renderItem'] = ({ item }) => {
-    return (
-      <View
-        style={[
-          styles.drawerElementContainer,
-          item.id === chapter.id && {
-            backgroundColor: color(theme.primary).alpha(0.12).string(),
-          },
-        ]}
-      >
-        <Pressable
-          android_ripple={{ color: theme.rippleColor }}
-          onPress={() => changeChapter(item)}
-          style={styles.chapterCtn}
-        >
-          <Text
-            numberOfLines={1}
-            style={[
-              styles.chapterNameCtn,
-              { color: item.unread ? theme.onSurface : theme.outline },
-            ]}
-          >
-            {item.name}
-          </Text>
-          {item.releaseTime ? (
-            <Text
-              style={[
-                styles.releaseDateCtn,
-                { color: item.unread ? theme.onSurfaceVariant : theme.outline },
-              ]}
-            >
-              {item.releaseTime}
-            </Text>
-          ) : null}
-        </Pressable>
-      </View>
-    );
-  };
-
-  const ListFooter = () => {
-    return (
-      <View style={styles.footer}>
-        <Button
-          mode="contained"
-          style={styles.button}
-          title={buttonProperties.up.text}
-          onPress={buttonProperties.up.func}
-        />
-        <Button
-          mode="contained"
-          style={styles.button}
-          title={buttonProperties.down.text}
-          onPress={buttonProperties.down.func}
-        />
-      </View>
-    );
-  };
-  const checkViewableItems: FlashListProps<ChapterInfo>['onViewableItemsChanged'] =
-    ({ viewableItems }) => {
-      let up = {
-        text: getString('readerScreen.drawer.scrollToTop'),
-        func: () => {
-          listRef.current?.scrollToIndex({ index: 0, animated: true });
-        },
-      };
-      let down = {
-        text: getString('readerScreen.drawer.scrollToBottom'),
-        func: () => {
-          listRef.current?.scrollToEnd({
-            animated: true,
-          });
-        },
-      };
-      if (viewableItems.length !== 0) {
-        let visible = viewableItems.find(({ item }) => {
-          return item.id === chapter.id;
+  const checkViewableItems = ({
+    viewableItems,
+  }: {
+    viewableItems: ViewToken[];
+  }) => {
+    let up = {
+      text: getString('readerScreen.drawer.scrollToTop'),
+      func: () => {
+        listRef.current?.scrollToIndex({ index: 0, animated: true });
+      },
```

Comment:
I think using index (undefinded if scroll to end)  instead of a callback is better.
```ts
const scroll = useCallback((index?: number) => {
  }, [listRef.current]);
```
Also, moving the ListFooter to this file and call `scroll callback` directly  in button oppress could be cleaner.

Reply:
I now refactored the file and I think I got what you meant. At least the file is now definitly cleaner.